### DISCUSSION TITLE
Audio

### DISCUSSION THREAD

**Discussion by Xact** [rabbibotton/clog#58] [Q&A] [answered]
Hi!
I'm loving this project - it has the potential to put common lisp on a great many radars - but am having a good old time trying to get it to move in my preferred direction: a highly responsive audio-playing app. I have a metronome playing, with adjustable tempo, but the playback is erratic... even after I coded a millisecond-perfect timer. So I suppose it will be necessary to go client-side to improve timing. And on this front (needing to use js) I have had real problems understanding what the heck is going on with clog and the interface with jquery/javascript. 

To illustrate my confusion: I have an audio "splash" script which plays on browser page launch (in autoplay permitting browsers). It works fine when I add the script to an element via the :content keyword. But I cannot get the script working with any combination of clog functions 'load-script' 'load-css' and 'js-execute'. Where should I direct my confusion... learning JQuery perhaps (if so... any recommended texts?) 

Attached is a very short script, which works AOK when added via the ':content' keyword: [foo.txt](https://github.com/rabbibotton/clog/files/6285016/foo.txt)

**Comment by rabbibotton** [comment]
Sorry last few week's holidays and work this slipped through the cracks, would have been better to file ticket. I'll try and look at this today or tomorrow.

**Comment by rabbibotton** [comment, marked as answer]
Is there a reason not using create-audio? Take a look at demo 1 which is a small game with sound.

**Reply by Xact** [reply]
I need to load audio client-side for performance. The html audio tag (which is what I assume jquery taps into) runs too hot, and too slow. I now have a page of js which loads audio client-side, and it all works nicely. The issue I have is that it seems a bit of a hack to have to incorporate it via the :content keyword to a clog element. Why can't I just incorporate it to the global clog body via load-script/js-execute and have it work that way? Anyhoo, I intend to use clog for the GUI as it seems a really nice way to do this, and I'm glad I can get my js to work, even if it is a "bit hacky". 

Thanks so much for all your work on CLOG - it really is inspiring.

**Reply by rabbibotton** [reply]
Exactly what you are looking for you can see in Demo 3, in do-ide-file-new for example. Using :content is as you say a bit hacky. If that still doesn’t work for you I’ll try and do a quick example.

**Reply by Xact** [reply]
ok thanks. I've been putting off an in-depth look at this demo for now. I guess it's time to have a closer look :)
